### ISSUE TITLE
crash--This problem still persists #37

### ISSUE THREAD

**Issue by cyqluowang** [TheFinestArtist/FinestWebView-Android#44]
```
03-14 23:53:36.931 18043 18043 E AndroidRuntime: Caused by: java.lang.ClassCastException: android.graphics.drawable.VectorDrawable cannot be cast to android.support.graphics.drawable.VectorDrawableCompat
03-14 23:53:36.931 18043 18043 E AndroidRuntime:    at com.thefinestartist.finestwebview.FinestWebViewActivity.updateIcon(FinestWebViewActivity.java:833)
03-14 23:53:36.931 18043 18043 E AndroidRuntime:    at com.thefinestartist.finestwebview.FinestWebViewActivity.layoutViews(FinestWebViewActivity.java:469)
03-14 23:53:36.931 18043 18043 E AndroidRuntime:    at com.thefinestartist.finestwebview.FinestWebViewActivity.onCreate(FinestWebViewActivity.java:859)
03-14 23:53:36.931 18043 18043 E AndroidRuntime:    at android.app.Activity.performCreate(Activity.java:6251)
03-14 23:53:36.931 18043 18043 E AndroidRuntime:    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
03-14 23:53:36.931 18043 18043 E AndroidRuntime:    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369)
```

**Comment by TheFinestArtist** [comment]
Does this solve your issue?
http://stackoverflow.com/a/35624744/1797648

**Comment by TheFinestArtist** [comment]
@cyqluowang 
Can you tell me little more information?
1. Which version of FinestWebView you are using
2. Gradle version
3. Support library version, especially appcompat-v7

**Comment by cyqluowang** [comment]
FinestWebView： 1.2.5
Gradle：com.android.tools.build:gradle:2.0.0
appcompat-v7：23.2.1

**Comment by cyqluowang** [comment]
```
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2392)
                                                                            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2443)
                                                                            at android.app.ActivityThread.access$800(ActivityThread.java:157)
                                                                            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1354)
                                                                            at android.os.Handler.dispatchMessage(Handler.java:110)
                                                                            at android.os.Looper.loop(Looper.java:193)
                                                                            at android.app.ActivityThread.main(ActivityThread.java:5348)
                                                                            at java.lang.reflect.Method.invokeNative(Native Method)
                                                                            at java.lang.reflect.Method.invoke(Method.java:515)
                                                                            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:829)
                                                                            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:645)
                                                                            at dalvik.system.NativeStart.main(Native Method)
                                                                         Caused by: java.lang.ClassCastException: android.graphics.drawable.BitmapDrawable cannot be cast to android.support.graphics.drawable.VectorDrawableCompat
                                                                            at com.thefinestartist.finestwebview.FinestWebViewActivity.updateIcon(FinestWebViewActivity.java:859)
                                                                            at com.thefinestartist.finestwebview.FinestWebViewActivity.layoutViews(FinestWebViewActivity.java:475)
                                                                            at com.thefinestartist.finestwebview.FinestWebViewActivity.onCreate(FinestWebViewActivity.java:885)
                                                                            at android.app.Activity.performCreate(Activity.java:5451)
                                                                            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1097)
                                                                            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2346)
                                                                            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2443) 
                                                                            at android.app.ActivityThread.access$800(ActivityThread.java:157) 
                                                                            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1354) 
                                                                            at android.os.Handler.dispatchMessage(Handler.java:110) 
                                                                            at android.os.Looper.loop(Looper.java:193) 
                                                                            at android.app.ActivityThread.main(ActivityThread.java:5348) 
                                                                            at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                            at java.lang.reflect.Method.invoke(Method.java:515) 
                                                                            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:829) 
                                                                            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:645) 
                                                                            at dalvik.system.NativeStart.main(Native Method) 

```

**Comment by cyqluowang** [comment]
```
E/dalvikvm: Could not find class 'android.telephony.CarrierConfigManager', referenced from method com.thefinestartist.utils.service.ServiceUtil.getCarrierConfigManager
04-19 15:00:16.956 32075-32075/com.renrun.qiantuhao E/dalvikvm: Could not find class 'android.hardware.fingerprint.FingerprintManager', referenced from method com.thefinestartist.utils.service.ServiceUtil.getFingerprintManager
04-19 15:00:16.956 32075-32075/com.renrun.qiantuhao E/dalvikvm: Could not find class 'android.app.job.JobScheduler', referenced from method com.thefinestartist.utils.service.ServiceUtil.getJobScheduler
04-19 15:00:16.956 32075-32075/com.renrun.qiantuhao E/dalvikvm: Could not find class 'android.content.pm.LauncherApps', referenced from method com.thefinestartist.utils.service.ServiceUtil.getLauncherApps
04-19 15:00:16.957 32075-32075/com.renrun.qiantuhao E/dalvikvm: Could not find class 'android.media.projection.MediaProjectionManager', referenced from method com.thefinestartist.utils.service.ServiceUtil.getMediaProjectionManager
04-19 15:00:16.957 32075-32075/com.renrun.qiantuhao E/dalvikvm: Could not find class 'android.media.session.MediaSessionManager', referenced from method com.thefinestartist.utils.service.ServiceUtil.getMediaSessionManager
04-19 15:00:16.957 32075-32075/com.renrun.qiantuhao E/dalvikvm: Could not find class 'android.media.midi.MidiManager', referenced from method com.thefinestartist.utils.service.ServiceUtil.getMidiManager
04-19 15:00:16.957 32075-32075/com.renrun.qiantuhao E/dalvikvm: Could not find class 'android.app.usage.NetworkStatsManager', referenced from method com.thefinestartist.utils.service.ServiceUtil.getNetworkStatsManager
04-19 15:00:16.958 32075-32075/com.renrun.qiantuhao E/dalvikvm: Could not find class 'android.content.RestrictionsManager', referenced from method com.thefinestartist.utils.service.ServiceUtil.getRestrictionsManager
04-19 15:00:16.958 32075-32075/com.renrun.qiantuhao E/dalvikvm: Could not find class 'android.telephony.SubscriptionManager', referenced from method com.thefinestartist.utils.service.ServiceUtil.getSubscriptionManager
04-19 15:00:16.958 32075-32075/com.renrun.qiantuhao E/dalvikvm: Could not find class 'android.telecom.TelecomManager', referenced from method com.thefinestartist.utils.service.ServiceUtil.getTelecomManager
04-19 15:00:16.958 32075-32075/com.renrun.qiantuhao E/dalvikvm: Could not find class 'android.media.tv.TvInputManager', referenced from method com.thefinestartist.utils.service.ServiceUtil.getTvInputManager
04-19 15:00:16.959 32075-32075/com.renrun.qiantuhao E/dalvikvm: Could not find class 'android.app.usage.UsageStatsManager', referenced from method com.thefinestartist.utils.service.ServiceUtil.getUsageStatsManager
04-19 15:00:16.975 240-741/? E/AudioFlinger: getNextBuffer, get null buffer
04-19 15:00:16.980 32075-32075/com.renrun.qiantuhao E/AndroidRuntime: FATAL EXCEPTION: main
                                                                      Process: com.renrun.qiantuhao, PID: 32075
                                                                      java.lang.RuntimeException: Unable to start activity ComponentInfo{com.renrun.qiantuhao/com.thefinestartist.finestwebview.FinestWebViewActivity}: java.lang.ClassCastException: android.graphics.drawable.BitmapDrawable cannot be cast to android.support.graphics.drawable.VectorDrawableCompat
                                                                          at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2392)
                                                                          at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2443)
                                                                          at android.app.ActivityThread.access$800(ActivityThread.java:157)
                                                                          at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1354)
                                                                          at android.os.Handler.dispatchMessage(Handler.java:110)
                                                                          at android.os.Looper.loop(Looper.java:193)
                                                                          at android.app.ActivityThread.main(ActivityThread.java:5348)
                                                                          at java.lang.reflect.Method.invokeNative(Native Method)
                                                                          at java.lang.reflect.Method.invoke(Method.java:515)
                                                                          at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:829)
                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:645)
                                                                          at dalvik.system.NativeStart.main(Native Method)
                                                                       Caused by: java.lang.ClassCastException: android.graphics.drawable.BitmapDrawable cannot be cast to android.support.graphics.drawable.VectorDrawableCompat
                                                                          at com.thefinestartist.finestwebview.FinestWebViewActivity.updateIcon(FinestWebViewActivity.java:859)
                                                                          at com.thefinestartist.finestwebview.FinestWebViewActivity.layoutViews(FinestWebViewActivity.java:475)
                                                                          at com.thefinestartist.finestwebview.FinestWebViewActivity.onCreate(FinestWebViewActivity.java:885)
                                                                          at android.app.Activity.performCreate(Activity.java:5451)
                                                                          at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1097)
                                                                          at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2346)
                                                                          at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2443) 
                                                                          at android.app.ActivityThread.access$800(ActivityThread.java:157) 
                                                                          at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1354) 
                                                                          at android.os.Handler.dispatchMessage(Handler.java:110) 
                                                                          at android.os.Looper.loop(Looper.java:193) 
                                                                          at android.app.ActivityThread.main(ActivityThread.java:5348) 
                                                                          at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                          at java.lang.reflect.Method.invoke(Method.java:515) 
                                                                          at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:829) 
                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:645) 
                                                                          at dalvik.system.NativeStart.main(Native Method) 
04-19 15:00:16.998 240-741/? E/AudioFlinger: getNextBuffer, get null buffer
```

**Comment by suliuyes** [comment]
Me too!!!
The same problem!!!

FinestWebView： 1.2.5
Gradle：com.android.tools.build:gradle:2.1.0
compile "com.android.support:appcompat-v7:23.2.1"
compile "com.android.support:design:23.2.1"
gradle-2.10-all.zip
vectorDrawables.useSupportLibrary = true

**Comment by suliuyes** [comment]
gradle-2.10-all.zip

**Comment by souravdas6** [comment]
Below API level 21, app is crashing for class ClassCastException: android.graphics.drawable.BitmapDrawable cannot be cast to android.support.graphics.drawable.VectorDrawableCompat                                                                       at com.thefinestartist.finestwebview.FinestWebViewActivity.updateIcon(FinestWebViewActivity.java:859)

**Comment by TheFinestArtist** [comment]
Seems like the vector compat support API has been removed temporarily. https://code.google.com/p/android/issues/detail?id=205236

I will update the library ASAP.

**Comment by TheFinestArtist** [comment]
Released 1.2.6 which doesn't uses any vector icons. I will reuse vector icons if the Android support library is ready.
